pkg/mutation/mutation_function_test.go

```diff
@@ -0,0 +1,234 @@
+package mutation_test
+
+import (
+	"fmt"
+	"testing"
+
+	mutationsv1alpha1 "github.com/open-policy-agent/gatekeeper/apis/mutations/v1alpha1"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const TestValue = "testValue"
+
+func prepareTestPod(t *testing.T) *unstructured.Unstructured {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testpod1",
+			Namespace: "foo",
+			Labels:    map[string]string{"a": "b"},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "testname1",
+					Ports: []corev1.ContainerPort{{Name: "portName1"}},
+				},
+				{
+					Name: "testname2",
+					Ports: []corev1.ContainerPort{
+						{Name: "portName2A"},
+						{Name: "portName2B"},
+					},
+				},
+				{
+					Name:  "testname3",
+					Ports: []corev1.ContainerPort{{Name: "portName3"}},
+				},
+			},
+		},
+	}
+	podObject, err := runtime.DefaultUnstructuredConverter.ToUnstructured(pod)
+	if err != nil {
+		t.Errorf("Failed to convert pod to unstructured %v", err)
+	}
+	return &unstructured.Unstructured{Object: podObject}
+}
+
+func TestObjects(t *testing.T) {
+
+	testFunc := func(unstructured *unstructured.Unstructured) {
+		labels := unstructured.Object["metadata"].(map[string]interface{})["labels"]
+		if labels.(map[string]interface{})["labelA"] != TestValue {
+			t.Errorf("Failed to update pod")
+		}
+
+	}
+
+	testMutation(
+		"metadata.labels.labelA",
```

Comment:
We are able to create a mutator object that touches `metadata.labels`?

Reply:
It looks like we disallow AssignMetadata with location other than "metadata. ...", but don't make any such check for Assign.
We should modify mutation.MutatorForAssign to check this.

Reply:
I thought I had seen a check on the Assign mutator that disallowed the `metadata` prefix. Is that that PR is not currently incorporated into this one?

Reply:
That's in the mutation validation pr

Reply:
Cool. We'll probably want to use Metadata mutators to mutate metadata for these tests, otherwise they'll fail when we integrate that pr.